internal/restic/snapshot_find.go

```diff
@@ -92,7 +97,7 @@ func FindFilteredSnapshots(ctx context.Context, repo Repository, host string, ta
 			return nil
 		}
 
-		if (host != "" && host != sn.Hostname) || !sn.HasTagList(tags) || !sn.HasPaths(paths) {
+		if (hosts != nil && !sn.HasHostname(hosts)) || !sn.HasTagList(tags) || !sn.HasPaths(paths) {
```

Comment:
Same here, the check for `nil` is not needed (and not entirely correct) as well.

Reply:
is changed.